docs/writing-tests.md

```diff
@@ -0,0 +1,99 @@
+# How to write tests with Eden
+
+The eden golang test SDK for eve consists of two categories of functions:
+
+1. `openevec`: infrastructure manager. `openevec` enables you to deploy a controller, a backend for that controller, deploy an edge node, etc. It does not change anything on an individual edge node.
+2. `evetestkit`: provides useful collection of functions to describe expected state of the system (controller, EVE, AppInstances)
+
+And those functions are used in context of standard golang test library. We do a setup in TestMain and then write test functions which interact with the environment we created. Setting up environment takes couple of minutes, so it makes sense to do it once and run tests within that environment. 
+
+Source for the example below can be found [here](../tests/sec/sec_test.go)
+
+In order to test a feature in Eden you need to
+
+## 1. Create a configuration file and describe your environment
+
+The glue between `openevec` and `evetestkit` is a configuration structrure called `EdenSetupArgs`. It contains all the necessary information to setup all the components of the test: controller, EVE, etc. You can fill in the structure manually, however `openevec` provides you with a convinient function to create default configuration structure providing project root path:
+
+```go
+// ... 
+func TestMain(m *testing.M) {
+	currentPath, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	twoLevelsUp := filepath.Dir(filepath.Dir(currentPath))
+
+	cfg := openevec.GetDefaultConfig(twoLevelsUp)
+
+	if err = openevec.ConfigAdd(cfg, cfg.ConfigName, "", false); err != nil {
+		log.Fatal(err)
+	}
+  // ... 
+}
+```
+
+Due to backward compatibility with escript as of time of writing, the project root path should be Eden repository folder. So if you add tests in the `eden/tests/my-awesome-test` you need to go two levels up. 
```

Comment:
I assume this also is planned to be removed, and will be once all of the escript tests are gone? If so, worth saying explicitly.